### PR TITLE
Geth chart refactoring

### PR DESCRIPTION
**Pull request by VladStarr**
- [x] migrate ingress to new apiVersion
- [x] improve p2p settings
- [x] add serviceMonitor CR
- [x] refactor default values

### CODE REVIEW

**Review by voron — APPROVED** [review]
LGTM, just some minor notes/questions

File: dysnix/geth/templates/service-rpc.yaml
Comment (VladStarr) [reply dysnix/geth/templates/service-rpc.yaml:40]: We can control this behaviour by using `.Values.podLabels` variable. I added it exactly with this reason

File: dysnix/geth/templates/_config.tpl
Comment (VladStarr) [reply dysnix/geth/templates/_config.tpl:62]: I think we can and it will be an improvement to security.
We will also need to add `chownDatadir` initContainer to start Geth from non-root user.

File: dysnix/geth/templates/scripts/_check-readiness.tpl
Comment (VladStarr) [reply dysnix/geth/templates/scripts/_check-readiness.tpl:6]: IMO it's not important to have ability of changing datadir for end-user. In k8s environment it does not seem to affect anything.

However, as I can see `geth attach` command simply ignores `--config` flag.

File: dysnix/geth/templates/service-rpc.yaml
Comment (voron) [reply dysnix/geth/templates/service-rpc.yaml:40]: Separate var should be a better option then hadrcoded label names.

File: dysnix/geth/templates/_config.tpl
Comment (VladStarr) [reply dysnix/geth/templates/_config.tpl:62]: You're right, as documentation says:
> By default, Kubernetes recursively changes ownership and permissions for the contents of each volume to match the fsGroup specified in a Pod's securityContext when that volume is mounted. For large volumes, checking and changing ownership and permissions can take a lot of time, slowing Pod startup. You can use the fsGroupChangePolicy field inside a securityContext to control the way that Kubernetes checks and manages ownership and permissions for a volume.

**Comment by VladStarr** [comment]
@voron, I made some extra changes, waiting for your approval.

File: dysnix/geth/values.yaml
Comment (voron) [inline dysnix/geth/values.yaml:32]: WDYT on keeping this comment in values.yaml ?

**Review by voron — APPROVED** [review]
Still LGTM :)

File: dysnix/geth/values.yaml
Comment (VladStarr) [reply dysnix/geth/values.yaml:32]: Yup i think i just missed it. Fixed.

**Pull request merged**
